site/src/pages/DeploySettingsPage/OAuth2AppsSettingsPage/OAuth2AppForm.tsx

```diff
@@ -0,0 +1,87 @@
+import LoadingButton from "@mui/lab/LoadingButton";
+import TextField from "@mui/material/TextField";
+import { type FC, type ReactNode } from "react";
+import { isApiValidationError, mapApiErrorToFieldErrors } from "api/errors";
+import type * as TypesGen from "api/typesGenerated";
+import { Stack } from "components/Stack/Stack";
+
+type OAuth2AppFormProps = {
+  app?: TypesGen.OAuth2App;
+  onSubmit: (data: {
+    name: string;
+    callback_url: string;
+    icon: string;
+  }) => void;
+  error?: unknown;
+  isUpdating: boolean;
+  actions?: ReactNode;
+};
+
+export const OAuth2AppForm: FC<OAuth2AppFormProps> = ({
+  app,
+  onSubmit,
+  error,
+  isUpdating,
+  actions,
+}) => {
+  const apiValidationErrors = isApiValidationError(error)
+    ? mapApiErrorToFieldErrors(error.response.data)
+    : undefined;
+
+  return (
+    <form
+      css={{ marginTop: 10 }}
+      onSubmit={(event) => {
+        event.preventDefault();
+        const formData = new FormData(event.target as HTMLFormElement);
+        onSubmit({
+          name: formData.get("name") as string,
```

Comment:
I get that the type assertions are necessary to exclude the `File` type, but do you think we should still have `null` checks?

Reply:
Honestly I should probably use Formik or whatever, and add the validations too.  I just...dislike Formik haha

I figure `null` will be caught by validation, either on the frontend (if I add it) or the backend.  But from a code complete perspective it does seem like adding `null` here is the right move.

Reply:
Since the page is blocked from being used outside of development for now I am going to punt frontend validation for another PR, if that seems chill.